src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,167 @@ full_subcategory_inclusion _
 
 end truncated
 
+section epi_mono
+
+/-- A morphism in `simplex_category` is a monomorphism precisely when it is an injective function
+-/
+theorem mono_iff_injective {n m : simplex_category} {f: n ⟶ m} :
+  (category_theory.mono f) ↔ (function.injective ⇑f.to_preorder_hom) :=
+begin
+  split,
+  { intros m x y h,
+  set cx : [0] ⟶ n := const n x with def_cx,
+  set cy : [0] ⟶ n := const n y with def_cy,
+  have H : (cx ≫ f = cy ≫ f) := by {dsimp, rw h},
+  resetI, rw (cancel_mono f) at H,
+  have from_constant_function :
+    (∀ (z : fin(n.len + 1)), (z = (const n z).to_preorder_hom.to_fun 0)),
+  { intro z, simp, refl},
+  rw [from_constant_function x, from_constant_function y, ←def_cx, ←def_cy, H] },
+  { intro H,
+  split,
+  intros Z g h hyp_eq,
+  dsimp,
+  ext,
+  apply_fun hom.to_preorder_hom at hyp_eq,
+  simp at hyp_eq,
+  apply_fun preorder_hom.to_fun at hyp_eq,
+  replace hyp_eq := (congr_fun hyp_eq x),
+  rw [preorder_hom.to_fun_eq_coe, preorder_hom.to_fun_eq_coe] at hyp_eq,
+  rw [preorder_hom.comp_coe, preorder_hom.comp_coe] at hyp_eq,
+  dsimp at hyp_eq,
+  rw H hyp_eq }
```

Comment:
If you'd like, you could show that the simplex category is concrete, then use general results to deduce this half:
```
instance : concrete_category.{0} simplex_category.{u} :=
{ forget :=
  { obj := λ i, fin (i.len + 1),
    map := λ i j f, f.to_preorder_hom },
  forget_faithful := {} }
```
then
```
  { intro hyp_inj,
    exact concrete_category.mono_of_injective _ hyp_inj }
```
(really this category is representably concrete so the other direction should come too, but we don't have that just yet!)